content/articles/online-machine-learning-with-river-python/index.md

```diff
@@ -0,0 +1,446 @@
+
+> NOTE: for you to follow along easily, use Google Colab
+
+### Introduction
+
+As mentioned earlier we shall use [River Python](https://riverml.xyz/latest/) to build our model. [River Python](https://riverml.xyz/latest/) is the best suited due to the following reasons.
+
+- Has incremental functionality
+  This library can be updated after each observation and therefore can be used to process streaming data.
+- It's adaptive
+  Since we are dealing with streaming data that keeps on changing over time, we need a library that is robust and can work under changing environments.
+
+- General-purpose
+  River Python is used for classification, clustering and regression. Both supervised and unsupervised machine learning.
+- Efficient & easy to use
+  It's very efficient when handling streaming data, it is also simple and easy to use and a beginner can easily follow.
+
+In online machine learning, the model uses real-time data fo training and makes a sigle observation at a time.
+We then update our model as each new data arrives, this increases the model accuracy over time. The input data is continuously used to improve the current model's knowledge and as we further train the model becomes more adaptive and robust.
+
+### River Python installation
+
+Since we are using Google Colab in this tutorial, we install using this command.
+
+```python
+!pip install river
+```
+
+#### Import River
+
+To use River we import it into our machine.
+
+```python
+import river
+```
+
+### Checking methods and attributes
+
+The River Python has various methods and attributes for online machine learning. To list all the available methods use the following command.
+
+```python
+dir(river)
+```
+
+List of River methods and attributes
+
+```bash
+['__all__',
+ '__builtins__',
+ '__cached__',
+ '__doc__',
+ '__file__',
+ '__loader__',
+ '__name__',
+ '__package__',
+ '__path__',
+ '__spec__',
+ '__version__',
+ 'anomaly',
+ 'base',
+ 'cluster',
+ 'compat',
+ 'compose',
+ 'datasets',
+ 'drift',
+ 'dummy',
+ 'ensemble',
+ 'evaluate',
+ 'expert',
+ 'facto',
+ 'feature_extraction',
+ 'feature_selection',
+ 'imblearn',
+ 'linear_model',
+ 'meta',
+ 'metrics',
+ 'multiclass',
+ 'multioutput',
+ 'naive_bayes',
+ 'neighbors',
+ 'neural_net',
+ 'optim',
+ 'preprocessing',
+ 'proba',
+ 'reco',
+ 'stats',
+ 'stream',
+ 'synth',
+ 'time_series',
+ 'tree',
+ 'utils']
+```
+
+Some of the methods that we will be using are as follows.
+
+- `naive_bayes`
+  This is the algorithm that we will be using to build our text classification model.
+
+- `preprocessing`
+  Will be used in the processing of our dataset used to train our model.
+
+- `metrics`
+  Calculate the accuracy score of our model.
+
+- `stream`
+  Will be used to simulate our dataset to be streaming data.
+
+- `anomaly`
+  Will be used to detect errors and anomalies in our model.
+
```

Comment:
You cannot start your sentences with the word 'will', please replace all these instances with, "It will...."